pkg/errors/conversion.go

```diff
@@ -15,10 +15,24 @@
 package errors
 
 import (
+	"context"
+	"net"
+
 	"github.com/golang/protobuf/proto"
 	"google.golang.org/grpc/status"
 )
 
+var (
+	errContextDeadlineExceeded = DefineDeadlineExceeded("context_deadline_exceeded", "context deadline exceeded")
+	errContextCanceled         = DefineCanceled("context_canceled", "context canceled")
+
+	errNetInvalidAddr    = DefineInvalidArgument("net_invalid_addr", "invalid address error: `{message}`", "message", "temporary", "timeout")
+	errNetAddr           = DefineUnavailable("net_addr", "address error: `{message}`", "message", "temporary", "timeout")
+	errNetDNS            = DefineUnavailable("net_dns", "DNS error: `{message}`", "message", "temporary", "timeout", "not_found")
+	errNetUnknownNetwork = DefineNotFound("net_unknown_network", "unknown network: `{message}`", "message", "temporary", "timeout")
+	errNetOperation      = DefineUnavailable("net_operation", "network operation error: `{message}`", "message", "op", "net", "source", "address", "timeout", "temporary")
```

Comment:
- We need the `error:` text in the error message
- You don't need to use backticks for the error message
- We can also consider just using the error message as-is, and use the error name to know what kind of error it is